src/AhRulesBot/AhRulesBot/BotRequestsProcessing/Handlers/CustomRulesMessageHandler.cs

```diff
@@ -0,0 +1,62 @@
+﻿using AhRulesBot.MessageProcessing.Interfaces;
+using AhRulesBot.Models;
+using Serilog;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Channels;
+
+namespace AhRulesBot.BotRequestsProcessing.Handlers
+{
+    internal class CustomRulesMessageHandler : IMessageHandler
+    {
+        private readonly IMessageHandler _next;
+        private readonly ChannelReader<List<CustomRuleItem>> _channel;
+        private readonly ILogger _logger;
+
+        private List<CustomRuleItem> CustomRules = new List<CustomRuleItem>();
+
+        public CustomRulesMessageHandler(
+            ILogger logger,
+            ChannelReader<List<CustomRuleItem>> channel,
+            IMessageHandler next)
+        {
+            _logger = logger;
+            _channel = channel;
+            _next = next;
+        }
+
+        public HandlerResult Handle(string message)
+        {
+            var command = TryParseAsCustomRulesRequest(message);
+            if (command)
+            {
+                return new HandlerResult { Data = ProcessRulesRequest(message) };
+            }
+
+            return _next != null ? _next.Handle(message) : new HandlerResult();
+        }
+
+        private List<string> ProcessRulesRequest(string message)
+        {
+            Func<CustomRuleItem, bool> messageContains = item => item.Title.Contains(message, StringComparison.InvariantCultureIgnoreCase);
+            Func<CustomRuleItem, bool> messageExact = item => item.Title.Equals(message, StringComparison.InvariantCultureIgnoreCase);
+
+            _channel.TryRead(out List<CustomRuleItem> rulesUpdate);
```

Comment:
I see potential issue here:
1. If users will not call this handler in over 5h + 5m - this channel would have two messages
2. Then, if some user would try to send some message calling this handler - user could receive different results (where first may be from 5 hours ago)

Reply:
Thank you! Fixed, could you please check?